app/models/user.rb

```diff
@@ -1,8 +1,9 @@
 class User < ApplicationRecord
   has_many :ratings
   validates :email, uniqueness: true
-  before_save :format_email
+  before_validation :format_email
```

Comment:
When an ActiveRecord object is persisted, it goes through it's callbacks in a [specific order](https://guides.rubyonrails.org/active_record_callbacks.html#creating-an-object). It's important to remember that the `before_save` hook happens _after_ validations are run.

Try this in `rails console` with the `main` branch checked out:

```
> User.create(name: "Levi", email: "Levi@jeans.com")
# now instantiate a user object, but don't persist it!
> u = User.new(name: "Levi", email: "Levi@jeans.com")
> u.valid?
```
The intended behavior is for `valid?` to be `false`. But you should be getting `true`

When the uniqueness validation runs, the email is still `Levi@jeans.com` because the `before_save` hasn't run yet.